frontend/src/api/calls/AssignmentCalls.ts

```diff
@@ -52,6 +67,40 @@ export async function deleteAssignment(
     return oldAssignments.filter((assignment) => assignment._links.self.href !== assignmentURL);
 }
 
+/**
+ * Function setting the assignment on a given url to invalid.
+ * @param assignmentUrl the url of the [IAssignment] that needs to be invalid.
+ */
+export async function invalidateAssignment(assignmentUrl: string): Promise<IAssignment | undefined> {
+    const response = await basePatch(assignmentUrl, { isValid: false });
+    const assignment = <IAssignment>response.data;
+    try {
+        // Check if response is a valid assignment.
+        assignment._links.self.href;
+        return assignment;
+    } catch (e) {
+        return undefined;
+    }
```

Comment:
Seems weird to use a try and catch here, can't you just look if assignment is defined? or of assignment._links is defined or something, as this is pretty ugly.